system/modules/core/library/Contao/User.php

```diff
@@ -381,7 +381,7 @@ public function login()
 		// The password has been generated with crypt()
 		if (\Encryption::test($this->password))
```

Comment:
In order to being able to deprecate the `Encryption::test()` method, we could use

```php
$info = password_get_info($this->password);

if ($info['algo'] > 0)
{
```

here. WDYT?

Reply:
Ah yes, wie could do that indeed. Then we can probably deprecate the whole `Encryption` class, can't we?

Reply:
Nope, just the hashing part. Encrypting and decrypting remains untouched.

Reply:
Talking about Contao 4.

Reply:
We do not (yet) have an alternative for encrypting and decrypting data in Contao 4.

Reply:
Of course we have, it's already documented? What are you talking about exactly?

Reply:
So how do you encrypt and decrypt data in Contao 4?

```php
$string = 'Hello world';
$encrypted = Encryption::encrypt($string);
$decrypted = Encryption::decrypt($encrypted);
```

Can you point me to where we have documented how to do this without the `Encryption` library?

Reply:
Ah, I see. The methods are deprecated without a concrete alternative. 😄

Reply:
Then yes, we can deprecate the whole class.

Reply:
`Using Encryption::encrypt() has been deprecated and will no longer work in Contao 5.0. Use a third-party library such as OpenSSL or phpseclib instead.` is a pretty well described alternative 😄 

So everything ready here or what do you need?